src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -946,59 +946,53 @@ int MacroAssembler::bitset_to_regs(unsigned int bitset, unsigned char* regs) {
   return count;
 }
 
-// Push lots of registers in the bit set supplied.  Don't push sp.
-// Return the number of words pushed
-int MacroAssembler::push_reg(unsigned int bitset, Register stack) {
+// Push lots of registers in the regset supplied.  Don't push sp.
+void MacroAssembler::push_reg(RegSet regset, Register stack) {
   DEBUG_ONLY(int words_pushed = 0;)
   CompressibleRegion cr(this);
 
   unsigned char regs[32];
-  int count = bitset_to_regs(bitset, regs);
+  int count = bitset_to_regs(regset.bits(), regs);
```

Comment:
I think it's better to have a zero-check for regs.bits() before we run into the loop in MacroAssembler::bitset_to_regs. So I would suggest make the old API "int MacroAssembler::push_reg(unsigned int bitset, Register stack)" a private member function and implement the new API as:

void MacroAssembler::push_reg(RegSet regset, Register stack) { if (regs.bits()) push_reg(regs.bits(), stack); }

Similar for pop_reg and other APIs here.

Reply:
Done, keep the old bitset verison and make them private.